webapp/shop/api/ua_contracts/advantage_mapper.py

```diff
@@ -186,3 +209,123 @@ def get_user_subscriptions(self, email: str) -> List[UserSubscription]:
             )
 
         return build_user_subscriptions(user_summary, listings)
+
+    def get_or_create_user_account(
+        self, marketplace, customer_info, captcha_value
+    ) -> Account:
+        try:
+            return self.get_purchase_account(marketplace)
```

Comment:
Does this call not need authentication?

Reply:
It does. If the user is logged in and user has a purchase account the `get_purchase_account` would return thier `Account` instance.

If user is not logged in or the user is logged in but has no purchase account, it will fall down the `except` path and we ensure an `Account` for them.

Reply:
Hmm so this relies on having set an attribute on the API class somewhere else?